discord/commands/core.py

```diff
@@ -1065,9 +1065,9 @@ def to_dict(self) -> Dict:
 
         return as_dict
 
-    def command(self, **kwargs) -> Callable[[Callable], SlashCommand]:
-        def wrap(func) -> SlashCommand:
-            command = SlashCommand(func, parent=self, **kwargs)
+    def command(self, cls: Type[T] = SlashCommand, **kwargs) -> Callable[[Callable], SlashCommand]:
+        def wrap(func) -> T:
+            command = cls(func, parent=self, **kwargs)
```

Comment:
Not too sure about the typing here

Reply:
A new `TypeVar` bound to `ApplicationContext` could be created for this.